### PR TITLE
Add comprehensive test coverage for EmptyExec and 4 other core modules

### PR DESCRIPTION
**Pull request by MAVRICK-1**
# Add comprehensive test coverage for EmptyExec and 4 other core modules

  Please be sure to look over the pull request guidelines here: https://github.com/spaceandtimelabs/sxt-proof-of-sql/blob/main/CONTRIBUTING.md#submit-pr.

  # Please go through the following checklist
  - [x] The PR title and commit messages adhere to guidelines here: https://github.com/spaceandtimelabs/sxt-proof-of-sql/blob/main/CONTRIBUTING.md. In particular `!` is used
   if and only if at least one breaking change has been introduced.
  - [x] I have run the ci check script with `source scripts/run_ci_checks.sh`.
  - [x] I have run the clean commit check script with `source scripts/check_commits.sh`, and the commit history is certified to follow clean commit guidelines as described
  here: https://github.com/spaceandtimelabs/sxt-proof-of-sql/blob/main/COMMIT_GUIDELINES.md
  - [x] The latest changes from `main` have been incorporated to this PR by simple rebase if possible, if not, then conflicts are resolved appropriately.

  # Rationale for this change

  Currently, the repo has 2292 uncovered lines of code. We should have 100% test coverage. This PR adds comprehensive test coverage for 5 core modules that previously had 0%
   test coverage.

  Contributes to #848.
/claim #848

  These modules are fundamental components of the proof-of-sql system:
  - `ParseError` is used throughout the codebase for error handling
  - `LiteralValue` handles all data types in SQL queries
  - `TableRef` is crucial for table reference parsing
  - `EmptyExec` is a core proof plan component
  - `log_memory_usage` is used extensively in proof evaluation

  # What changes are included in this PR?

  - Add 8 tests for `ParseError` enum covering error creation, display, equality, cloning, and trait implementations
  - Add 13 tests for `LiteralValue` enum covering all variants (Boolean, integers, strings, binary, decimal, scalar, timestamp) and their conversions
  - Add 25 tests for `TableRef` struct covering all constructors, parsing methods, traits, and edge cases
  - Add 10 tests for `EmptyExec` covering all ProofPlan trait methods and evaluation functions
  - Add 7 tests for `log_memory_usage` utility function ensuring it handles various inputs gracefully

  Total: 63 new tests covering ~740 lines of test code.

  # Are these changes tested?

  Yes. All 63 tests pass successfully:
  
  

https://github.com/user-attachments/assets/76845aeb-1d42-4688-a112-23433cf222c5

  ```bash
  RUSTFLAGS="-Clink-arg=-fuse-ld=gold" BLITZAR_BACKEND=cpu cargo test --lib -- error::tests literal_value::tests table_ref::tests empty_exec::tests log::tests
  # Result: 63 passed; 0 failed

  The tests themselves are the changes in this PR. They provide comprehensive coverage including:
  - All public API methods
  - Edge cases and boundary values
  - Error conditions
  - Trait implementations (Debug, Clone, PartialEq, Serialize/Deserialize)
  - Empty/null input handling

### CODE REVIEW

**Comment by MAVRICK-1** [comment]
cc @JayWhite2357

File: crates/proof-of-sql/src/base/database/literal_value.rs [hunk 1]
```diff
@@ -85,3 +85,228 @@ impl LiteralValue {
...
+        assert_eq!(bigint_val.column_type(), ColumnType::BigInt);
+        assert_eq!(int128_val.column_type(), ColumnType::Int128);
+
+        // Test scalar conversion
+        let _uint8_scalar: TestScalar = uint8_val.to_scalar();
+        let _tinyint_scalar: TestScalar = tinyint_val.to_scalar();
+        let _smallint_scalar: TestScalar = smallint_val.to_scalar();
+        let _int_scalar: TestScalar = int_val.to_scalar();
+        let _bigint_scalar: TestScalar = bigint_val.to_scalar();
+        let _int128_scalar: TestScalar = int128_val.to_scalar();
```
[the remark this answers is not in the document]
Comment (MAVRICK-1) [reply crates/proof-of-sql/src/base/database/literal_value.rs]: @iajoiner Yes, I fixed it. The changes I made:

  1. Removed underscore prefixes - Changed from let _uint8_scalar to let uint8_scalar so the variables are actually used
  2. Added assertions - Added assert_eq! statements to verify each scalar conversion produces the expected value
  3. Verified correctness - Each assertion checks that to_scalar() converts the literal value to the correct scalar representation

  The test now properly validates that the scalar conversions work correctly, which addresses the reviewer's concern about "doing more than just converting them to scalars.
